cmd/operator-sdk/scorecard/cmd.go

```diff
@@ -65,5 +78,107 @@ func NewCmd() *cobra.Command {
 	if err := viper.BindPFlag("scorecard."+scorecard.BundleOpt, scorecardCmd.Flags().Lookup(scorecard.BundleOpt)); err != nil {
 		log.Fatalf("Unable to add bundle :%v", err)
 	}
+
 	return scorecardCmd
 }
+
+func initConfig() (*viper.Viper, error) {
+	// viper/cobra already has flags parsed at this point; we can check if a config file flag is set
+	if viper.GetString(scorecard.ConfigOpt) != "" {
+		// Use config file from the flag.
+		viper.SetConfigFile(viper.GetString(scorecard.ConfigOpt))
+	} else {
+		viper.AddConfigPath(projutil.MustGetwd())
+		// using SetConfigName allows users to use a .yaml, .json, or .toml file
+		viper.SetConfigName(scorecard.DefaultConfigFile)
+	}
+
+	var scViper *viper.Viper
+	if err := viper.ReadInConfig(); err == nil {
+		scViper = viper.Sub("scorecard")
+		// this is a workaround for the fact that nested flags don't persist on viper.Sub
+		scViper.Set(scorecard.OutputFormatOpt, viper.GetString("scorecard."+scorecard.OutputFormatOpt))
+		scViper.Set(scplugins.KubeconfigOpt, viper.GetString("scorecard."+scplugins.KubeconfigOpt))
+		scViper.Set(schelpers.VersionOpt, viper.GetString("scorecard."+schelpers.VersionOpt))
+		scViper.Set(scorecard.SelectorOpt, viper.GetString("scorecard."+scorecard.SelectorOpt))
+		scViper.Set(scorecard.BundleOpt, viper.GetString("scorecard."+scorecard.BundleOpt))
+		scViper.Set(scorecard.ListOpt, viper.GetString("scorecard."+scorecard.ListOpt))
+		// configure logger output before logging anything
+		if !scViper.IsSet(scorecard.OutputFormatOpt) {
+			scViper.Set(scorecard.OutputFormatOpt, scorecard.TextOutputFormat)
+		}
+		format := scViper.GetString(scorecard.OutputFormatOpt)
+		if format == scorecard.TextOutputFormat {
+			scorecard.LogReadWriter = os.Stdout
+		} else if format == scorecard.JSONOutputFormat {
+			scorecard.LogReadWriter = &bytes.Buffer{}
+		} else {
+			return nil, fmt.Errorf("invalid output format: %s", format)
+		}
+		scorecard.Log.SetOutput(scorecard.LogReadWriter)
+		if err != nil {
+			return nil, err
+		}
+		scorecard.Log.Info("Using config file: ", viper.ConfigFileUsed())
+	} else {
+		return nil, fmt.Errorf("could not read config file: %v\nSee %s for more information about the scorecard config file", err, scorecard.ConfigDocLink())
+	}
+	return scViper, nil
+}
+
+func buildScorecardConfig(c *scorecard.Config) {
+
+	scViper, err := initConfig()
+	if err != nil {
+		log.Printf("%v", err.Error())
+		os.Exit(1)
+	}
+
+	outputFormat := scViper.GetString(scorecard.OutputFormatOpt)
+	if outputFormat != scorecard.TextOutputFormat && outputFormat != scorecard.JSONOutputFormat {
+		log.Printf("%v", fmt.Errorf("invalid output format (%s); valid values: %s, %s", outputFormat, scorecard.TextOutputFormat, scorecard.JSONOutputFormat))
```

Comment:
Use `github.com/sirupsen/logrus` for logging, then call:
```suggestion
		log.Fatalf("invalid output format (%s); valid values: %s, %s", outputFormat,
		  scorecard.TextOutputFormat, scorecard.JSONOutputFormat)
```
And remove `os.Exit(1)`. Ditto below.

Reply:
changed to logrus and Fatalf in next commit.  Good suggestion.